zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -120,29 +131,185 @@ public static Either<ProblemDetail, ProcessInstanceSort> toSearchQuerySort(
+        } else {
+          // Collect validation error for unknown sortBy field
+          validationErrors.add(
+              ProblemDetail.forStatusAndDetail(
+                  HttpStatus.BAD_REQUEST, "Unknown sortBy field: " + field));
+          continue;
+        }
+
+        if ("asc".equalsIgnoreCase(order)) {
+          builder.asc();
+        } else if ("desc".equalsIgnoreCase(order)) {
+          builder.desc();
+        } else {
+          // Collect validation error for unknown sortOrder
+          validationErrors.add(
+              ProblemDetail.forStatusAndDetail(
+                  HttpStatus.BAD_REQUEST, "Unknown sortOrder: " + order));
+        }
+      }
+
+      if (!validationErrors.isEmpty()) {
+        return Either.left(
+            validationErrors.get(0)); // Return the first encountered error for simplicity
+      }
+
+      return Either.right(builder.build());
+    }
+
+    return Either.right(null);
+  }
+
+  public static UserTaskFilter toUserTaskFilter(final UserTaskFilterRequest filter) {
+    final var builder = FilterBuilders.userTask();
+
+    if (filter != null) {
+      // key
+      if (filter.getKey() != null) {
+        builder.userTaskKeys(filter.getKey());
+      }
+
+      // state
+      if (filter.getTaskState() != null && !filter.getTaskState().isEmpty()) {
+        builder.userTaskState(filter.getTaskState());
+      }
+
+      // assignee
+      if (filter.getAssignee() != null && !filter.getAssignee().isEmpty()) {
+        builder.assignees(filter.getAssignee());
+      }
+
+      // candidateGroup
+      if (filter.getCandidateGroup() != null && !filter.getCandidateGroup().isEmpty()) {
+        builder.candidateGroups(filter.getCandidateGroup());
+      }
+
+      // candidateUser
+      if (filter.getCandidateUser() != null && !filter.getCandidateUser().isEmpty()) {
+        builder.candidateUsers(filter.getCandidateUser());
+      }
+
+      // processDefinitionKey
+      if (filter.getProcessDefinitionKey() != null) {
+        builder.processDefinitionKeys(filter.getProcessDefinitionKey());
+      }
+
+      // processInstanceKey
+      if (filter.getProcessInstanceKey() != null) {
+        builder.processInstanceKeys(filter.getProcessInstanceKey());
+      }
+
+      // tenantIds
+      if (filter.getTenantIds() != null) {
+        builder.tenantIds(filter.getTenantIds());
+      }
+
+      // creationTime
+      if (filter.getCreationTime() != null) {
+        builder.creationDate(toDateValueFilter(filter.getCreationTime()));
+      }
+
+      // completionTime
+      if (filter.getCompletionTime() != null) {
+        builder.completionDate(toDateValueFilter(filter.getCompletionTime()));
+      }
+
+      // dueDate
+      if (filter.getDueDate() != null) {
+        builder.dueDate(toDateValueFilter(filter.getDueDate()));
+      }
+
+      // followUpDate
+      if (filter.getFollowUpDate() != null) {
+        builder.followUpDate(toDateValueFilter(filter.getFollowUpDate()));
+      }
```

Comment:
🔧 I guess those calls can run into exceptions when the date Strings cannot be parsed. Do we have any tests for this? If those calls run into exceptions, we need to catch them and create validation errors, just like in the other methods here.

Reply:
I have update this on the follow commits: 
- https://github.com/camunda/camunda/pull/19625/commits/a8c8a4b72fa4854b0cc70d1bec06dcddc4da2f05 
- https://github.com/camunda/camunda/pull/19625/commits/55bd74cf1ca948c641ad37081c3cf2675c85209f